nautobot/extras/models/models.py

```diff
@@ -484,6 +522,74 @@ def clean(self):
         if self.local_context_data and type(self.local_context_data) is not dict:
             raise ValidationError({"local_context_data": 'JSON data must be in object form. Example: {"foo": 123}'})
 
+        if self.local_context_schema and not self.local_context_data:
+            raise ValidationError({"local_context_schema": "Local context data must exist for a schema to be applied."})
+
+        # Validate data against schema
+        self._validate_with_schema("local_context_data", "local_context_schema")
+
+
+@extras_features(
+    "custom_fields",
+    "custom_validators",
+    "graphql",
+    "relationships",
+)
+class ConfigContextSchema(OrganizationalModel):
+    """
+    This model stores jsonschema documents where are used to optionally validate config context data payloads.
+    """
+
+    name = models.CharField(max_length=200, unique=True)
+    description = models.CharField(max_length=200, blank=True)
+    slug = models.SlugField()
+    data_schema = models.JSONField(
+        help_text="A JSON Schema document which is used to validate a config context object."
+    )
+    # A ConfigContextSchema *may* be owned by another model, such as a GitRepository, or it may be un-owned
+    owner_content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        limit_choices_to=FeatureQuery("config_context_owners"),
+        default=None,
+        null=True,
+        blank=True,
+    )
+    owner_object_id = models.UUIDField(default=None, null=True, blank=True)
+    owner = GenericForeignKey(
+        ct_field="owner_content_type",
+        fk_field="owner_object_id",
+    )
+
+    def __str__(self):
+        return self.name
+
+    def get_absolute_url(self):
+        return reverse("extras:configcontextschema", args=[self.slug])
+
+    def clean(self):
+        """
+        Validate the schema
+        """
+        super().clean()
+
+        try:
+            Draft7Validator.check_schema(self.data_schema)
+        except SchemaError as e:
+            raise ValidationError({"data_schema": e.message})
+
+        if (
+            type(self.data_schema) is not dict
```

Comment:
Is there a specific reason to not use `not isinstance(self.data_schema, dict)` here?